gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+Created: 2024-10-21
+Updated: 2024-10-21
+Stage: Draft
+Discussions-To: <TODO>
+Category: Protocol Logic
+Depends-On: GIP-0066, GIP-0068
+Replaces: GIP-0058
+---
+
+## Abstract
+
+This GIP describes **Indexing Payments**, a mechanism by which a protocol participant (commonly a gateway) can incentivize a candidate indexer to service a subgraph. It details a target, trust-minimized end-state that relies on Graph Horizon, while also introducing a semi-trusted Minimum Viable Product (MVP) based on the current protocol and off-chain components (gateway + indexer stacks). This GIP is an alternative, simplified proposal to [GIP-0058](https://github.com/graphprotocol/graph-improvement-proposals/blob/main/gips/0058-Replacing-Bonding-Curves-with-Indexing-Fees.md).
+
+## Motivation
+
+Curation has generally been perceived as a source of friction for Developers that may only want to have subgraphs indexed to power their decentralized applications. It requires locking up certain amounts of GRT, and the necessary amounts might vary depending on curation on other subgraphs. While the quality of service provided on the network is generally high, Indexers collect rewards for posting Proofs of Indexing independently to the query service they provide, so Developers are not guaranteed a certain level of service for the amount of curation they put into the system.
+
+For Indexers, there is no clear relationship between the Indexing Rewards on a subgraph and the amount of work required to index it. Attempts at defining better curation and rewards models to pay for indexing have so far been unsuccessful, which presents a need for a simpler mechanism where parties interested in indexing a subgraph can simply pay for it through the protocol. However, when building such a mechanism, it is imperative to preserve the protocol's trust-minimization and decentralization principles.
+
+## Prior Art
+
+GIP-0058 presents a previous version of a very similar proposal. This GIP re-introduces the same concepts, but incorporates the latest approach in Graph Horizon, and simplifies the proposal with an off-chain MVP while clarifying that the actor engaging in **Indexing Payments** is generally a Gateway running automated indexer selection.
+
+## High-Level Description
+
+We are proposing a new coordination mechanism to incentivize subgraph serving to a performant level. Such a coordination mechanism is necessary to jump-start, and sometimes maintain, performant subgraph serving. One such time is when a subgraph is being upgraded. In early conversations this has been referred to as **Direct Indexing Payments** (DIPs).
+
+In particular, the coordination mechanism has been designed with the following in mind:
+
+- The objective is to get subgraphs served through the protocol to a sufficiently performant level
+- The price to get a subgraph served at said level should ideally be related to the amount of work required.
```

Comment:
I'll second, although I picked up earlier too. It would be fine if you have a broader goal that includes quality of service that this mechanism explicitly defines as a goal (clarifying what serving to a performant lever means) and addresses. It might be you talk about this later and I should come back and update this comment.

Reply:
Reading onwards I see that quality of service is a factor. This is valid and great to make part of the narrative IMO.

I would consider making the first point about being able to select indexers that best meet consumer needs, in terms of price and quality of service. You might consider then adding a separate point on incentivising indexers to provide good quality of service (due to them being able to attract more work). That might more directly link to what the proposal suggests.

Reply:
Sufficiently performant level could be reworded as sufficient level of performance

Reply:
Ok, I've re-worded the GIP to address this issue. Let me know what you think.

tldr; I've removed mentions of "level of performance" and "quality of service" in favor of "subgraph serving" (which is defined at the beginning) and "subgraph serving characteristics" (for which I've given examples but haven't defined)